Demo/RCHSearchViewController.swift

```diff
@@ -52,94 +78,109 @@ class RCHSearchViewController: UIViewController, UISearchBarDelegate, UICollecti
+        // Temp: Config API for usable API key
+        
+        guard let currentUserID = UserDefaults.standard.string(forKey: kRCHUserDefaultKeyCurrentUser) else {
+            fatalError()
+        }
         
-        // TODO: Replace with search call when ready
-        let strategyBuilder: RCHStrategyRecsBuilder = RCHSDK.builderForRecs(with: .siteWideBestSellers)
+        let config = RCHAPIClientConfig(apiKey: "showcaseparent", apiClientKey: "199c81c05e473265", endpoint: RCHEndpointProduction, useHTTPS: false)
+        config.apiClientSecret = "r5j50mlag06593401nd4kt734i"
+        config.userID = currentUserID
+        config.sessionID = UUID().uuidString
         
-        RCHSDK.defaultClient().sendRequest(strategyBuilder.build(), success: { (responseObject) in
+        RCHSDK.defaultClient().configure(config)
+    }
+
+    func searchForProducts() {
+        let placement: RCHRequestPlacement = RCHRequestPlacement.init(pageType: .search, name: "find")
+        let searchBuilder: RCHSearchBuilder = RCHSDK.builder(forSearch: placement, withQuery: searchTerm)
+        
+        RCHSDK.defaultClient().sendRequest(searchBuilder.build(), success: { (responseObject) in
             
-            guard let strategyResponse = responseObject as? RCHStrategyResult else {
-                print("Result Error")
+            guard let searchResult = responseObject as? RCHSearchResult else {
                 return
             }
-            guard let productArray = strategyResponse.recommendedProducts else {
-                print("Recommended Products Error")
-                return
-            }
-            self.productArray = productArray
             
-            if self.productArray.count == 0 {
-                self.showNoResults()
-            } else {
-                self.searchResultsCollectionView.isHidden = false
-                self.searchResultsCollectionView?.reloadData()
+            self.products = searchResult.products!
 
-            }
         }) { (responseObject, error) in
             print(error)
         }
-//        resetSearch()
     }
     
-    func resetSearch() {
-        view.endEditing(true)
-//        autocompleteArray = []
-        autocompleteTableView.reloadData()
-        autocompleteTableView.isHidden = true
-        searchProductsView.isHidden = false
+    func autocomplete(withQuery text: String) {
+        let autocompleteBuilder: RCHAutocompleteBuilder = RCHSDK.builderForAutocomplete(withQuery: text)
+        
+        RCHSDK.defaultClient().sendRequest(autocompleteBuilder.build(), success: { (responseObject) in
+            
+            guard let responseAutocompleteSuggestions = responseObject as? [RCHAutocompleteSuggestion] else {
+                print("Result Error")
+                return
+            }
+            
+            self.autocompleteSuggestions = responseAutocompleteSuggestions.map({$0.text!})
+        }) { (responseObject, error) in
+            print(error)
```

Comment:
We should present an alert if there's an error with the `localizedDescription` of the error as the message.